extern/storage-sealing/fsm.go

```diff
@@ -391,6 +393,37 @@ func (m *Sealing) plan(events []statemachine.Event, state *SectorInfo) (func(sta
 	return nil, processed, nil
 }
 
+func (m *Sealing) onUpdateSector(ctx context.Context, state *SectorInfo) error {
+	if m.getConfig == nil {
+		return nil // tests
+	}
+
+	cfg, err := m.getConfig()
+	if err != nil {
+		return xerrors.Errorf("getting config: %w", err)
+	}
+	sp, err := m.currentSealProof(ctx)
+	if err != nil {
+		return xerrors.Errorf("getting seal proof type: %w", err)
+	}
+
+	shouldUpdateInput := m.stats.updateSector(cfg, m.minerSectorID(state.SectorNumber), state.State)
+
+	// trigger more input processing when we've dipped below max sealing limits
+	if shouldUpdateInput {
+		go func() {
+			m.inputLk.Lock()
+			defer m.inputLk.Unlock()
+
+			if err := m.updateInput(ctx, sp); err != nil {
```

Comment:
Could we end up triggering an avalanche of input updates (there's no deduping)? I assume nothing is going to really happen until we call `updateInput` but I figured I'd check.

Reply:
This should only be getting called when we  go from 'too many sectors sealing' to 'can seal more sectors', which should happen once per sealed sector, which means that even if you are a really, really big miners, you likely won't call this more than a few times per epoch (and at that scale it's highly likely that you're not even using this code)